.github/workflows/ci-release.yml

```diff
@@ -0,0 +1,43 @@
+name: Release Builder
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+jobs:
+  build-jdk8:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Check NO SNAPSHOT version suffix
+        run: if [[ $(cat gradle.properties | grep version= | sed 's/^version=//') =~ .*-SNAPSHOT ]]; then exit 1; else exit 0; fi
+      - name: Set up JDK 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build with Gradle
+        env:
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGKEY }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGPASSWORD }}
+          SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
+          SONATYPE_TOKEN: ${{ secrets.SONATYPE_TOKEN }}
+          CI: true
+        run: ./gradlew --parallel --max-workers=4 -PreleaseBuild=true clean check && ./gradlew -PreleaseBuild=true publish
+      - name: Publish Test Results
+        if: ${{ always() }}
+        uses: scacap/action-surefire-report@v1
```

Comment:
for 3rd party actions is better to use a commit hash instead of a tag. 
tags are not immutable and a bad actor could change Action code, harvest secrets, etc..

Reply:
thank @jerrinot! I will submit a followup PR to address this and make it consistent for other actions as well.

Reply:
https://github.com/apple/servicetalk/pull/1271